syz-fuzzer/fuzzer.go

```diff
@@ -282,8 +282,8 @@ func (tool *FuzzerTool) startExecutingCall(progID int64, pid, try int) {
 	})
 }
 
-func (tool *FuzzerTool) exchangeDataCall(needProgs int, results []rpctype.ExecutionResult,
-	latency time.Duration) time.Duration {
+func (tool *FuzzerTool) exchangeDataCall(results []rpctype.ExecutionResult, latency time.Duration) time.Duration {
+	needProgs := max(0, cap(tool.requests)/2-len(tool.requests))
```

Comment:
It shouldn't happen during normal work, right.
Now with the new version where I added a timer, it may happen during checking. But we can use it to exchange other data (stats/signal), it's not only programs.
But really I don't see how to fix it well in the current architecture. Again, with flatrpc it will all be much simpler and better (we can just push messages both ways as needed).